### PR TITLE
[WIP] Dependency injection tests refactor

### PR DESCRIPTION
**Pull request by Soullivaneuh**
### Subject

Let's use [matthiasnoback/symfony-dependency-injection-test](https://github.com/matthiasnoback/SymfonyDependencyInjectionTest) to make tests cleaner and easier.
### To do
- [x] Bundle tests
- [x] Configuration tests
- [ ] Extension tests
- [ ] Test issue proof of #3909 and add a not to **never** put legacy for `addClassToCompile` test

### CODE REVIEW

**Comment by Soullivaneuh** [comment]
Test on `SonataAdminExtension` are failing because it gets the `kernel.bundles` that does not exist here.

I will open an issue for that, but I think we can avoid the parameter usage.

Maybe with class/method detection?

cc @greg0ire @sonata-project/contributors

File: composer.json [hunk 1]
```diff
@@ -49,7 +49,8 @@
         "sllh/php-cs-fixer-styleci-bridge": "^2.0",
         "sonata-project/intl-bundle": "^2.2.4",
         "symfony/phpunit-bridge": "^2.7 || ^3.0",
-        "symfony/yaml": "^2.3 || ^3.0"
+        "symfony/yaml": "^2.3 || ^3.0",
+        "matthiasnoback/symfony-dependency-injection-test": "^0.7.6"
```
Comment (greg0ire) [inline composer.json]: Please sort dependencies

Comment (Soullivaneuh) [inline composer.json]: Would be great to have the `sort-packages` option.

**Comment by core23** [comment]
Can you please fix travis @Soullivaneuh

File: composer.json (hunk 1 above)
Comment (OskarStark) [inline composer.json]: https://github.com/sonata-project/SonataAdminBundle/blob/3.x/composer.json#L73

is already done

**Comment by OskarStark** [comment]
any news on this PR?

**Comment by greg0ire** [comment]
> Test on SonataAdminExtension are failing because it gets the kernel.bundles that does not exist here.

This is not a problem @Soullivaneuh , simply set the parameter by hand, that's what I did in #4029

**Comment by core23** [comment]
@Soullivaneuh Can you give a status update?

**Comment by core23** [comment]
This PR looks dead. Feel free to reopen if you will continue working on this.

**Pull request closed without merging**